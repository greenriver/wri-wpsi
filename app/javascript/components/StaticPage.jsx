import React from 'react'
import injectSheet from 'react-jss'

const styles = {
  main: {
    height: '100%',
    overflow: 'auto',
  },
  contentContainer: {
    maxWidth: '800px',
    margin: '20px auto',
    padding: '20px',
    flex: '1 1 auto',
  },
}

const StaticPage = ({ page, classes }) => {
  return (
    <main className={classes.main}>
      <div className={classes.contentContainer}>
        <content dangerouslySetInnerHTML={{__html: page.content}} />
      </div>
    </main>
  )
}

import PropTypes from 'prop-types'
StaticPage.propTypes = {
  page: PropTypes.exact({
    name: PropTypes.string.isRequired,
    slug: PropTypes.string.isRequired,
    content: PropTypes.string.isRequired,
  }).isRequired,
  classes: PropTypes.object.isRequired,
}

export default injectSheet(styles)(StaticPage)
