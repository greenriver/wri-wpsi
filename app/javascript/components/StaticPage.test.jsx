import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import StaticPage from 'components/StaticPage'

const page = {
  name: 'Test name',
  slug: 'Test slug',
  content: 'Test content',
}
const component = shallow(<StaticPage page={page} />)

it('renders correctly', () => {
  let tree = toJson(component)
  expect(tree).toMatchSnapshot()
})
