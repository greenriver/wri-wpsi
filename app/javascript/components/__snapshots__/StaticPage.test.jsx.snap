// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<StaticPage
  classes={
    Object {
      "contentContainer": "StaticPage-contentContainer-0-1-2",
      "main": "StaticPage-main-0-1-1",
    }
  }
  page={
    Object {
      "content": "Test content",
      "name": "Test name",
      "slug": "Test slug",
    }
  }
/>
`;
